auth0/src/test/java/com/auth0/android/authentication/storage/SecureCredentialsManagerTest.java

```diff
@@ -750,6 +751,36 @@ public void shouldGetCredentialsAfterAuthentication() {
         assertThat(retryCheck, is(false));
     }
 
+    @Test
+    public void shouldNotGetCredentialsWhenCredentialsHaveExpired() {
+        Date credentialsExpiresAt = new Date(CredentialsMock.CURRENT_TIME_MS + 123456L * 1000);
+        Date storedExpiresAt = new Date(CredentialsMock.CURRENT_TIME_MS - 60L * 1000);
+        insertTestCredentials(true, true, false, credentialsExpiresAt);
+        when(storage.retrieveLong("com.auth0.credentials_expires_at")).thenReturn(storedExpiresAt.getTime());
+
+        //Require authentication
+        Activity activity = spy(Robolectric.buildActivity(Activity.class).create().start().resume().get());
+        KeyguardManager kService = mock(KeyguardManager.class);
+        when(activity.getSystemService(Context.KEYGUARD_SERVICE)).thenReturn(kService);
+        when(kService.isKeyguardSecure()).thenReturn(true);
+        Intent confirmCredentialsIntent = mock(Intent.class);
+        when(kService.createConfirmDeviceCredentialIntent("theTitle", "theDescription")).thenReturn(confirmCredentialsIntent);
+        boolean willRequireAuthentication = manager.requireAuthentication(activity, 123, "theTitle", "theDescription");
+        assertThat(willRequireAuthentication, is(true));
+
+        manager.getCredentials(callback);
+
+        //Should fail because of expired credentials
+        verify(callback).onFailure(exceptionCaptor.capture());
+        CredentialsManagerException exception = exceptionCaptor.getValue();
+        assertThat(exception, is(notNullValue()));
+        assertThat(exception.getMessage(), is("No Credentials were previously set."));
```

Comment:
Shouldn't the message convey the fact that the credentials have expired, instead of saying that there were no credentials?

Reply:
No. Having expired credentials is the same as not having them, as the developer won't be able to obtain them. This is the same response given when the user checks for "valid credentials" with the corresponding method.